File: 3886-count-number-of-trapezoids-i/count-number-of-trapezoids-i.go
func countTrapezoids(points [][]int) int {
	const MOD = 1e9 + 7
	mp := map[int]int{}
	for _, p := range points {
		mp[p[1]]++
	}
    
	seg := []int{}
	for _, v := range mp {
		if v >= 2 {
			seg = append(seg, v*(v-1)/2%MOD)
		}
	}

	sum, ans := 0, 0
	for _, v := range seg {
		ans = (ans + v*sum) % MOD
		sum = (sum + v) % MOD
	}

	return ans
}

File: 3886-count-number-of-trapezoids-i/count-number-of-trapezoids-i.java
class Solution {
    public int countTrapezoids(int[][] points) {
        int n = points.length;
        int mod = 1000000007;

        HashMap<Long, Long> hm = new HashMap<>();
        for (int i = 0; i < n; i++)
            hm.put((long) points[i][1], hm.getOrDefault((long) points[i][1], 0L) + 1);

        List<Long> al = new ArrayList<Long>();
        for (long key : hm.keySet())
            if (hm.get(key) >= 2)
                al.add(((hm.get(key) * (hm.get(key) - 1)) / 2) % mod);

        long total = 0;

        long ans = 0;
        for (int i = 0; i < al.size(); i++) {
            ans = (ans + al.get(i) * total % mod) % mod;
            total = (total + al.get(i)) % mod;
        }

        return (int) ans;
    }
}